bot/main.py

```diff
@@ -73,6 +77,15 @@ class DBEntry(TypedDict):
     view: bool
 
 
+class CheckFailedException(commands.CommandError):
+    """Custom exception to help identify which check function failed."""
+
+    def __init__(self, check_name: str):
+        """Initialise with the name of the check."""
+        super().__init__()
+        self.check_name = check_name
+
+
```

Comment:
Instead of storing the "exception type as a string" I would suggest a more OOP-like approach

```suggestion
class CheckAdminOrAcademic(commands.CheckFailure):
    pass

class CheckAdmin(CheckAdminOrAcademic):
    pass

class CheckAcademic(CheckAdminOrAcademic):
    pass
```

This way `isinstance(error, CheckAdmin)` can detect admin command failure, `isinstance(error, CheckAcademic)` can detect academic server error, and `isinstance(error, CheckAdminOrAcademic)` can be used to detect either of above